W-19476402 Add parameters to tool reference (#11)
mulesoft/docs-mulesoft-mcp-server: modules/ROOT/pages/reference-mcp-tools.adoc

Available only in Anypoint Code Builder.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|mode
|The operation to perform on the agent network project YAML. Valid values are `configure` and `edit`.

For example, use `configure` to set up a custom agent network tailored to your environment, discover agent network components, or scaffold a new agent network. Use `edit` to make minor changes to a configured agent network YAML file, such as renaming an existing agent.
|Yes

|prompt
|Prompt from the user containing instructions to the agent.
|Yes

|===

[[create-agent-network-project]]
== create_agent_network_project

Create and scaffold an agent network project with a name and location you specify.

Available only in Anypoint Code Builder.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|projectName
|Unique name of the project. The project name is used as the name of the agent network project file.
|Yes

|projectPath

> |Parent location of the project. The default is the parent folder of the current workspace. The project will reside in a subfolder of projectPath.

Reviewer: projectPath? Is it "the project path" or `projectPath`?

|Yes

|projectDescription
|Optional description of the project.
|No

|groupId
|Optional organization or business group ID of the project.
|No

|===

[[create-and-manage-api-instances]]
== create_and_manage_api_instances

Create and configure API instances in Anypoint API Manager.

[%header,cols="1,3,1"]
|===
|Parameter |Description | Required?

|organizationId
|The ID of the organization where the instance resides. Default is the ID of the user`s organization from the current session.
|No

|environmentName

> |The name of the environment where the instance resides. Default is the environment for the organization specified in organizationId.

Reviewer: ```suggestion
|The name of the environment where the instance resides. Default is the environment for the organization specified in `organizationId`.
```

Create an asset in Anypoint Exchange, update some Anypoint Exchange asset properties without republishing the asset, and delete an asset in Anypoint Exchange. Asset properties are the asset name, description, icon, contact name, and contact email.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|operation
|Operation to perform on the asset. Use `create` to create a new asset, `update` to modify an existing asset, or `delete` to remove an asset.
|Yes

|assetName
|Name of the asset.
|Yes

|assetDescription
|Description of the asset.
|No

|assetIcon
|Icon for the asset.
|No

|contactName
|Contact name for the asset.
|No

|contactEmail
|Contact email for the asset.
|No

|===

[[create-and-manage-metadata]]
== create_and_manage_metadata

Define metadata for a flow component, attach metadata to Mule flow components, and update existing metadata. To define metadata, provide the Mule configuration file URI and sample data or schema. To apply metadata, specify existing metadata types. To update metadata, provide new schema or sample data.

Available only in Anypoint Code Builder.

[%header,cols="1,3a,1"]
|===
|Parameter |Description |Required?

|projectPath

> |Path of the existing Mule project where pom.xml is located.

Reviewer: ```suggestion
|Path of the existing Mule project where the `pom.xml` is located.
```

|Yes

|operation
|Task to perform. Valid values are:

* `define` to create metadata.
* `attach` to apply metadata to flow components.
* `update` to update existing metadata.
|Yes

|metadata
|String value of sample data or schema JSON object. JSON original format is required (no conversion from other formats). 
|No

Required when the operation parameter value is `define`.

|dataFormat
|The format of the metadata being processed. Specify `JSON` for JSON-formatted metadata or `XML` for XML-formatted metadata.
|No

|metadataPath
|Path of the sample metadata or schema file. 
|No

Required when the operation parameter value is `define`.

|componentNodeDocId

> |The doc:id attribute value for the flow component node. Don't provide the child node's doc:id.

Reviewer: ```suggestion
|The `doc:id` attribute value for the flow component node. Don't provide the child node's `doc:id`.
```

|No

> Required when the operation parameter value is `attach`.

Reviewer: ```suggestion
Required when the operation's parameter value is `attach`.
```

|metadataName

> |Name of the new metadata to be defined based on a sample data JSON, for example, `customer` if sample data is for a customer. 

Reviewer: ```suggestion
|Name of the new metadata to define based on a sample data JSON, for example, `customer` if sample data is for a customer. 
```

|No

Required when the operation parameter value is `define`.

|isSchema
|Whether the metadata is a schema.
|No

>  Required when the operation parameter value is `define`.

Reviewer: ```suggestion
 Required when the operation's parameter value is `define`.
```

Reviewer: ? I think these should be possessive. But if I'm wrong, ignore.

|===

[[create-api-spec-project]]
== create_api_spec_project

> Create and scaffold an API spec or fragment project. You can specify the project name, location, and spec language (RAML, OAS, etc.).

Reviewer: ```suggestion
Create and scaffold an API spec or fragment project. Specify the project name, location, and spec language (RAML or OAS).
```

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|projectName
|Name of the API spec project.
|Yes

|projectLocation
|Directory path where the project will be created.
|Yes

|specLanguage
|Specification language for the API project. Use `RAML` for RESTful API Modeling Language or `OAS` for OpenAPI Specification.
|Yes

|===

[[create-mcp-server]]
== create_MCP_server

> Create a MuleSoft MCP server. This tool handles requests for creating MCP servers using the Anypoint Connector for MCP (MCP Connector).

Reviewer: ```suggestion
Create a MuleSoft MCP server. This tool handles requests for creating MCP servers using Anypoint Connector for MCP (MCP Connector).
```

Prerequisite: Install the xref:mcp-connector::index.adoc[MCP Connector].

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|projectPath
|Path where the Mule project folder is located.
|Yes

|projectXmlPath
|Path to the Mule project XML file.
|Yes

|===

[[create-mule-project]]
== create_mule_project

Create an integration project.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|projectName
|Name of the Mule project.
|Yes

|projectLocation
|Directory path where the project will be created.
|Yes

|muleVersion
|Mule runtime version for the project.
|No

|===

[[deploy-agent-network]]
== deploy_agent_network

> Deploy an agent network instance to a runtime environment. Specify the project path, target environment, deployment target, and Managed Flex Gateway configuration for deployment. Prerequisites: Managed Flex Gateway, a business group, and the agent network project contains a valid `exchange.json` file. 

Reviewer: ```suggestion
Deploy an agent network instance to a runtime environment. Specify the project path, target environment, deployment target, and Managed Flex Gateway configuration for deployment. 

Prerequisites: 

* A Managed Flex Gateway
* A business group
* A valid `exchange.json` file in the agent network project 
```

Reviewer: I might be incorrect about this, but I thought agent network requires at least one small managed Flex Gateway and one large Managed Flex Gateway. I looked in the doc and don't see that mentioned, so I'm not sure where I got that idea. Or at least two gateways (ingress and egress) and that might be wrong for this context.

Author: I think that's part of the configuration specifics.

|Path to the agent network project to deploy.
|No

|environmentName
|Target environment name (prompts for selection if not provided).
|No

|privateSpaceName
|Private space name (prompts for selection if not provided).
|No

|ingressGatewayName
|Ingress gateway name (prompts for selection if not provided).
|No

|egressGatewayName
|Egress gateway name (prompts for selection if not provided).
|No

|===

[[deploy-mule-application]]
== deploy_mule_application

Deploy your application to CloudHub 2.0 or Runtime Fabric and run it. You can deploy from a local project or Anypoint Exchange asset. This tool supports security, high availability, and performance optimization deployment options.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|applicationName
|Name of the application to deploy.
|Yes

|projectPath
|Path to the Mule project directory.
|No

|target
|Deployment target environment. Use `CloudHub-US` for CloudHub US region, `CloudHub-EU` for CloudHub EU region, or `RuntimeFabric` for Runtime Fabric deployment.
|Yes

|replicas
|Number of replicas for the application.
|No

|vCores
|Number of vCores to allocate.
|No

|===

[[generate-api-spec]]
== generate_api_spec

> Generate, modify, or enhance an API spec using AI. Prerequisites: Log in to Anypoint Code Builder with your Anypoint Platform credentials. Make sure Einstein is enabled in Access Management.

Reviewer: ```suggestion
Generate, modify, or enhance an API spec using AI. 

Prerequisites: 

* Log in to Anypoint Code Builder with your Anypoint Platform credentials. 
* Make sure Agentforce is enabled in Access Management.
```

Reviewer: Maybe xref to Einstein enablement? They actually rebranded Einstein to Agentforce, so it's access-management::enabling-agentforce.adoc

[%header,cols="1,3a,1"]
|===
|Parameter |Description |Required?

|language
|Language of the API specification to generate. Use `oas` for OpenAPI specifications or `raml` for RAML specifications.
|Yes

|languageVersion
|

* If `language=oas`, valid value is `3.0`.
* If `language=raml`, valid value is `1.0`.
|Yes

|specSyntax
|Syntax of the generated API spec.

* If `language=oas` or `oas-components`, valid values are `json` and `yaml`.
* If `language=raml` or `raml-fragments`, valid value is `raml`.
|Yes

|projectType
|Valid value is `API_SPEC`.
|Yes

|naturalLanguagePrompt
|Input to the Large Language Model (LLM).
|Yes

|===

[[generate-mule-flow]]
== generate_mule_flow

Create a flow using AI.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|flowDescription
|Description of the flow to generate.
|Yes

|projectPath
|Path to the Mule project where the flow will be created.
|No

|===

[[get-flex-gateway-policy-example]]
== get_flex_gateway_policy_example

> Get a Flex Gateway policy example to help you configure your Rust source code. You can get example policies that manipulate a request's header and body, make HTTP and gRPC calls to external services, share data between requests and responses, reject requests, perform sleep operations, and the use of PDK's libraries.

Reviewer: Rust source code? That might be correct if it's referring to the programming language.

Author: Yes.

Author: https://docs.mulesoft.com/pdk/latest/policies-pdk-configure-features

[%header,cols="1,3a,1"]
|===
|Parameter |Description |Required?

|feature
|The type of policy feature to generate an example for. Valid values are:

* `authentication` to propagate authentication information to other policies.
* `body_manipulation` to read or update request/response body.
* `body_stream` to read request/response body by chunks.
* `cache` to use the cache library to share data across workers.
* `contracts` to authenticate and authorize against the management plane.

> * `control_flow` to stop request flow and generate a custom response.

Reviewer: ```suggestion
* `control_flow` to stop the request flow and generate a custom response.
```

* `cors` to use the Cross-Origin Resource Sharing library.
* `dataweave` to define and evaluate custom DataWeave expressions.

> * `grpc` to make gRPC call to external service.

Reviewer: ```suggestion
* `grpc` to make a gRPC call to an external service.
```

> * `header_manipulation` to perform CRUD for request/response headers.

Reviewer: ```suggestion
* `header_manipulation` to perform CRUD operations for request/response headers.
```

> * `http_call` to make HTTP call to external service.

Reviewer: ```suggestion
* `http_call` to make an HTTP call to an external service.
```

> * `jwt` to parse and validate JSON Web Token.

Reviewer: ```suggestion
* `jwt` to parse and validate JSON Web Tokens.
```

* `lock` to synchronize workers.
* `logger` to print custom messages.

> * `metadata` to access metadata of the policy, api, and environment.

Reviewer: ```suggestion
* `metadata` to access metadata of the policy, API, and environment.
```

* `policy_violation` to mark requests as violations in monitoring traffic.
* `request_data` to share data between requests and responses.
* `timer` to perform sleep operations.
* `worker_variable` to share data across requests of a single worker.
|Yes

|===

[[get-platform-insights]]
== get_platform_insights

> See how your organization leverages the Anypoint Platform through usage and operational insights. These include:

Reviewer: ```suggestion
See how your organization leverages Anypoint Platform through usage and operational insights. These include:
```

* Usage trends for apps, such as flow count, message counts, and data throughput by business group and environment. Usage trends are for the current month. If it`s early in the month, you get usage for the last full month.

> * API and app call volume across consumers for last 7 days.

Reviewer: ```suggestion
* API and app call volume across consumers for the last 7 days.
```

> * API and app performance metrics such as errors and latency for last 7 days.

Reviewer: ```suggestion
* API and app performance metrics such as errors and latency for the last 7 days.
```

Reviewer: Maybe "the" isn't necessary.

|Yes

.3+s|Policy Management
|<<get-flex-gateway-policy-example,get_flex_gateway_policy_example>>
|No

|<<manage-api-instance-policy,manage_api_instance_policy>>
|No

|<<manage-flex-gateway-policy-project,manage_flex_gateway_policy_project>>
|No

.2+s|Usage and Insights
|<<get-platform-insights,get_platform_insights>>
|No

|<<get-reuse-metrics,get_reuse_metrics>>
|No

|===

[[add-governance-ruleset]]
== add_governance_ruleset

Add a governance ruleset from Anypoint Exchange to your project.
Use the `search_asset` tool to find a ruleset that matches an assetId, groupId, and version from user-provided criteria. The `search_asset` tool uses `searchQuery='user'` criteria and `assetFilters=ruleset` to return a ruleset meeting those criteria. Then, this tool adds the ruleset as a dependency to your project.

Available only in Anypoint Code Builder.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|assetId
|The ID of the asset.
|Yes

|groupId
|Group ID of the asset in Anypoint Exchange.
|Yes

|version
|Version of the asset, for example, `1.4.0`.
|Yes

|===

[[configure-agent-network-yaml]]
== configure_agent_network_yaml

> Modifies an agent network project YAML file. The tool can replace placeholder YAML created from a template with values generated from user prompts and guided setup (mode='configure'). Or, the tool can make minor changes to a preconfigured agent network YAML file (mode='edit'). The LLM determines the appropriate mode value from user prompts and guided setup.

Reviewer 2: ```suggestion
Modifies an agent network project YAML file. The tool can replace placeholder YAML created from a template with values generated from user prompts and guided setup (`mode='configure'`). Or, the tool can make minor changes to a preconfigured agent network YAML file (`mode='edit'`). The LLM determines the appropriate mode value from user prompts and guided setup.
```

|Yes

|projectDescription
|Description of the project.
|No

|groupId
|Organization or business group ID of the project.
|No

|===

[[create-and-manage-api-instances]]
== create_and_manage_api_instances

Create and configure API instances in Anypoint API Manager.

[%header,cols="1,3a,1"]
|===
|Parameter |Description | Required?

|organizationId
|The ID of the organization where the instance resides. The default is the ID of the user's organization from the current session.
|No

|environmentName
|The name of the environment where the instance resides. The default is the environment for the organization specified in `organizationId`.
|No

|apiInstanceId
|The instance ID of the API.
|No

Required if you're configuring an API instance.

|apiInstanceLabel
|Label or name of the API instance.
|No

|operation
|Operation to perform on the API instance. Valid values are:

* `create` to create an API instance.
* `configureBasicEndpointInMuleApplication` to configure a basic endpoint in a Mule application.
|Yes

|groupId
|Group ID of the API asset in Anypoint Exchange.
|No

|assetId
|ID of the API asset in Anypoint Exchange. Value can be the API name. 

> |Yes

Reviewer 2: Should this be No since it is only "required to create an API instance"?

Required to create an API instance.

|assetVersion
|Version of the API asset in Anypoint Exchange.
|No

|runtimeEngine
|Engine the API instance runs on. Valid value is `mule4`.
|No

|muleProxyType
|Valid value is `basicEndpoint`.
|No

|endpointUri
|Endpoint URI of the API instance (upstream).
|No

Required if `muleProxyType=basicEndpoint`.

|consumerEndpoint
|Consumer endpoint of the API instance.
|No

|providerId
|Provider ID of the API instance.
|No

|===

[[create-and-manage-assets]]
== create_and_manage_assets

Create, update, and delete assets in Anypoint Exchange.

[%header,cols="1,3a,1a"]
|===
|Parameter |Description |Required?

|operation
|Operation to perform on the asset. Valid values are:

* `create` to create an asset.
* `update` to modify an existing asset.
* `delete` to remove an asset.
|Yes

|groupId
|Group ID of the asset in Anypoint Exchange.
|No

|assetId
|The ID of the asset.
|No

|version

> |Version of the asset. Example: 1.4.0.

Reviewer 2: ```suggestion
|Version of the asset, for example, `1.4.0`.
```

|No

Required for asset creation.

|organizationId

> |The ID of the org to publish the org to.

Reviewer 2: ```suggestion
|The ID of the org to publish the asset to.
```

* `soft-delete` indicates logical deletion.
* `hard-delete` indicates physical deletion.

The default value is `soft-delete`.
|No

|contactName
|New contact name for the asset. Used for updating the asset.
|No

|contactEmail
|New contact email of the asset. Used for updating the asset.
|No

|iconFile
|Path to the icon file (image formats such as PNG and JPG) to add to the asset. Used for updating the asset.
|No

|===

[[create-and-manage-metadata]]
== create_and_manage_metadata

Define metadata for a flow component, attach metadata to Mule flow components, and update existing metadata. To define metadata, provide the Mule configuration file URI and sample data or schema. To apply metadata, specify existing metadata types. To update metadata, provide new schema or sample data.

Available only in Anypoint Code Builder.

[%header,cols="1,3a,1a"]
|===
|Parameter |Description |Required?

|projectPath
|Path of the Mule project directory where the `pom.xml` file is located.
|Yes

|operation
|Task to perform. Valid values are:

* `define` to create metadata.
* `attach` to apply metadata to flow components.
* `update` to update existing metadata.
|Yes

|metadata

> |String value of sample data or schema JSON object. JSON original format is required (no conversion from other formats). If this metadata is for input or output, use the applyMetadataOn parameter.

Reviewer 2: ```suggestion
|String value of sample data or schema JSON object. JSON original format is required (no conversion from other formats). If this metadata is for input or output, use the `applyMetadataOn` parameter.
```

|No

Required when the operation parameter value is `define`.

|dataFormat
|The format of the metadata to process. Specify `JSON` for JSON-formatted metadata or `XML` for XML-formatted metadata.
|No

|metadataPath
|Path of the sample metadata or schema file. To specify if this metadata is for input or output, use the `applyMetadataOn` parameter.
|No

Required when the operation parameter value is `define`.

|componentNodeDocId
|The `doc:id attribute` value of the flow component. For example, if the node is `<ee:transform doc:name='Transform' doc:id='hdipfo' >`, specify `hdipfo` as the `componentNodeDocId`. 
|No

> * Required when the operation is `apply`

Reviewer 2: ```suggestion
* Required when the operation is `apply`.
```

|No

Required when the operation parameter value is `update`.

|toolSummary
|Description of the tool.
|No

|===

[[create-mule-project]]
== create_mule_project

Create a Mule project structure with optional parameters, but no flow generation.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|projectPath
|Directory to create project in (defaults to the current workspace).
|No

|projectName
|Name for the project (defaults to a generated name).
|No

|===

[[create-api-spec-project]]
== create_api_spec_project

Create and scaffold an API spec or fragment project. You can specify the project name, location, and spec language (RAML or OAS).

[%header,cols="1,3a,1"]
|===
|Parameter |Description |Required?

|projectName
|Name of the API spec project.
|Yes

|projectLocation
|Directory path in which to create the project.
|Yes

|specLanguage
|Specification language for the API project. Valid values are:

* `RAML` for RESTful API Modeling Language
* `OAS` for OpenAPI specification
|Yes

|===

[[create-mcp-server]]
== create_MCP_server

Create a MuleSoft MCP server. This tool handles requests for creating MCP servers using Anypoint Connector for MCP (MCP Connector).

> Prerequisite: Install the xref:mcp-connector::index.adoc[MCP Connector].

Reviewer 2: ```suggestion
Prerequisite: Install xref:mcp-connector::index.adoc[MCP Connector].
```

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|projectPath
|Path location of the Mule project folder.
|Yes

|projectXmlPath
|Path to the Mule project XML file.
|Yes

|===

[[dataweave-create-documentation]]
== dataweave_create_documentation

Generates DataWeave documentation using the proper syntax and returns the content for manual insertion into the project.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|documentation
|Detailed description of what the function does.
|Yes

|parameters
|Any parameters applicable to the function.
|Yes

|example
|An example of how to use the function.
|Yes

|===

[[dataweave-create-project]]
== dataweave_create_project

> Creates a new DataWeave project, returns the URL of the created project created, and prompts to open the project.

Reviewer 2: ```suggestion
Creates a new DataWeave project, returns the URL of the created project, and prompts to open the project.
```

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|groupId
|The group ID of the project.
|No

|projectName
|The project name.
|Yes

|url
|The URL location where you want to create the new project. The default, if not specified, is `{USER_HOME}/dw-projects/`.
|No

|===

[[dataweave-create-sample-data]]
== dataweave_create_sample_data

Create DataWeave input sample data for a given DataWeave transformation file and return the path where it was created. For example, use this tool to create sample data for a variable if a preview fails with `There is no variable named 'variableName'`.

Prerequisite: Create the DataWeave script file before running `dataweave_create_sample_data`.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|url

> |The URL of DataWeave transformation file to create sample input for. The file must be in the format of `file:://path`. For example, `\file:///home/mdeachaval/test.dwl`. By default, uses the currently open file.

Reviewer 2: ```suggestion
|The URL of the DataWeave transformation file to create sample input for. The file must be in the format of `file:://path`. For example, `\file:///home/mdeachaval/test.dwl`. By default, uses the currently open file.
```

|Yes

|content
|The content of the sample input.
|Yes

|extension
|The file extension for the sample content file type. For example `json` for a JSON file or `xml` for an XML file. The default value is `json`.
|Yes

|name
|The name of the input to generate, such as `payload`. The default value is `payload`.
|Yes

|===

[[dataweave-get-module-metadata]]
== dataweave_get_module_metadata

Returns the list of all functions available in a DataWeave module, with documentation for each function, the input parameter types, and the output types. It includes the list of labels for each function to use when searching.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|url
|The URL of the project in the format `\file://<path>`.
|Yes

|moduleUrl

> |The URL of the DataWeave module file in the format `\file://<path>.dwl` or `jar://<path>.dwl` search documentation for. Use the `dataweave_get_project_metadata` tool to get the URL of a given module.

Reviewer 2: ```suggestion
|The URL of the DataWeave module file in the format `\file://<path>.dwl` or `jar://<path>.dwl` to search documentation for. Use the `dataweave_get_project_metadata` tool to get the URL of a given module.
```

* `body_stream` to read request/response body by chunks.
* `cache` to use the cache library to share data across workers.
* `contracts` to authenticate and authorize against the management plane.
* `control_flow` to stop the request flow and generate a custom response.
* `cors` to use the Cross-Origin Resource Sharing library.
* `dataweave` to define and evaluate custom DataWeave expressions.
* `grpc` to make a gRPC call to an external service.
* `header_manipulation` to perform CRUD operations for request/response headers.
* `http_call` to make an HTTP call to an external service.
* `jwt` to parse and validate JSON Web Tokens.
* `lock` to synchronize workers.
* `logger` to print custom messages.
* `metadata` to access metadata of the policy, API, and environment.
* `policy_violation` to mark requests as violations in monitoring traffic.
* `request_data` to share data between requests and responses.
* `timer` to perform sleep operations.
* `worker_variable` to share data across requests of a single worker.
|Yes

|===

[[get-platform-insights]]
== get_platform_insights

See how your organization leverages Anypoint Platform through usage and operational insights. These include:

> * Usage trends for apps, such as flow count, message counts, and data throughput by business group and environment. Usage trends are for the current month. If it`s early in the month, you get usage for the last full month.

Reviewer 2: ```suggestion
* Usage trends for apps, such as flow count, message counts, and data throughput by business group and environment. Usage trends are for the current month. If it's early in the month, you get usage for the last full month.
```

|Whether to include P99 latency in the response time. The default value is `false`.
|No

|latencyPercentile
|If `includeLatency=true`, this parameter specifies which percentile to use for latency measurements. Valid values are:

* `P99` for 99th percentile
* `P95` for 95th percentile
* `P90` for 90th percentile
* `P75` for 75th percentile 

The default value is `P99`.
|No

|===

[[get-reuse-metrics]]
== get_reuse_metrics

Get visibility into assets reuse across your organization. Drive efficiency, consistency, and cost savings by reducing duplication. Includes sandbox and production environments. Metrics are a year-to-date snapshot.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|includeReuseMetrics
|Whether the output includes reuse metrics for applications. Default is false.
|No

|===

[[get-topics-agent-actions-metadata-instructions]]
== get_topics_agent_actions_metadata_instructions

Get metadata and instructions for enabling topics and agent actions from an OpenAI spec. Topics are bundles of actions and sets of instructions agents use to complete tasks. Actions are tasks that the agent can do, and instructions help the agent understand how to use actions.

> Available only in Anypoint Code Builder.

Reviewer 2: is table missing?

Author: No parameters! Clarified. https://gus.lightning.force.com/lightning/r/ADM_Work__c/a07EE00002Lsy2JYAR/view

[[list-api-instances]]
== list_api_instances

Get information for instances in an environment. You can also filter by an instance ID or label and include applied or available policies. Use the `page` parameter to  paginate results.

Prompt tip: To get information for all instances, revise results using the `page` parameter.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|organizationId
|The ID of the organization where instances reside. Default is the ID of the default organization.
|No

|environmentName
|The name of the environment where instances reside. Default is the name of the default environment for the organization specified in `organizationId`.
|No

|apiInstanceId

> |

Reviewer 2: desc missing?

|No

Required if `includeAvailablePolicies=true`.

|apiInstanceLabel
|Label or name of the API instance.
|No

|page
|Page number of the results. Each page contains information for up to 10 instances.
|Yes

|includeAppliedPolicies
|Whether to include applied policies in results.
|No

|includeAvailablePolicies

> |Whether to include available policies in results. Requires apiInstanceId.

Reviewer 2: ```suggestion
|Whether to include available policies in results. Requires `apiInstanceId`.
```

|hasPerformanceOptimization
|Whether to enable performance optimization. The default is false.
|No

|isSecure
|Whether to enable advanced security features. The default is false.
|No

|===

[[validate-api-against-ruleset]]
== validate_api_against_ruleset

Validate an API specification in your project against governance rulesets or rules that you specify. If you specify all rulesets, the tool uses an empty `rules` parameter to validate all rulesets. Similarly, if you specify no ruleset, the tool uses an empty `rules` parameter. To find rules or rulesets that you specify, this tool calls the `list_governance_rulesets` tool.

Available only in Anypoint Code Builder.

[%header,cols="1,3,1"]
|===
|Parameter |Description |Required?

|projectPath
|Path where the project folder is located. The default is the current workspace. For absolute paths, use `file:` followed by three slash characters.
|No

|rules
|Array of API rules to validate. If empty, the tool validates all rulesets in `projectPath`.
|No

|syncCentralizedRulesets
|Whether the tool syncs centralized rulesets before validating the API specification. The default value is true.
|No

|===

[[validate-project]]
== validate_project

> Validate the current project. For example, an API spec or an agent network project. This tool doesn't validate against API Governance rulesets. To validate against those rulesets, use the `validate_api_against_rules` tool.

Reviewer 2: ```suggestion
Validate the current project. For example, an API spec or an agent network project. This tool doesn't validate against API Governance rulesets. To validate against those rulesets, use the `validate_api_against_ruleset` tool.
```

Author: Good catch!
